src/CommunityCommons/javasource/communitycommons/actions/ReverseList.java

```diff
@@ -0,0 +1,51 @@
+// This file was generated by Mendix Studio Pro.
+//
+// WARNING: Only the following code will be retained when actions are regenerated:
+// - the import list
+// - the code between BEGIN USER CODE and END USER CODE
+// - the code between BEGIN EXTRA CODE and END EXTRA CODE
+// Other code you write will be lost the next time you deploy the project.
+// Special characters, e.g., é, ö, à, etc. are supported in comments.
+
+package communitycommons.actions;
+
+import com.mendix.systemwideinterfaces.core.IContext;
+import com.mendix.webui.CustomJavaAction;
+import com.mendix.systemwideinterfaces.core.IMendixObject;
+
+/**
+ * This Java action is able to take a generic input list from any entity, and return a reversed list for the list entered in the parameters.
+ */
+public class ReverseList extends CustomJavaAction<java.util.List<IMendixObject>>
+{
+	private java.util.List<IMendixObject> InputList;
+
+	public ReverseList(IContext context, java.util.List<IMendixObject> InputList)
+	{
+		super(context);
+		this.InputList = InputList;
+	}
+
+	@java.lang.Override
+	public java.util.List<IMendixObject> executeAction() throws Exception
+	{
+		// BEGIN USER CODE
+        for(int i = 0, j = InputList.size() - 1; i < j; i++) {
+            InputList.add(i, InputList.remove(j));
```

Comment:
Hi Reinout, I used this as a reference https://www.techiedelight.com/reverse-list-java-inplace/, which indeed states both as an option. Personally I thought this was a bit clearer because you could see exactly what was going on instead of passing it on to another library, and I wasn't exactly sure on what the impact would be of importing another library into the community commons. But I'm not a Java expert by any means, and both are definitely a possibility

Reply:
As @reinouts stated, it is not necessary to reimplement it while Java Standart Lib already contains such a function. Additionally Java version reverses the list by swapping when applicable. This way is much efficient in terms of time and space complexity.

Reply:
Agreed with with is stated above that the option to use the Java standard library is logical in this context. We don't think we would add a lot by simply adding a wrapper to that in Community Commons.